storage/innobase/handler/ha_innodb.cc

```diff
@@ -3909,22 +3909,23 @@ during startup, will be encrypted with tablespace key which has empty UUID
 bool
 innobase_fix_tablespaces_empty_uuid()
 {
+	if (Encryption::master_key_id == 0) {
```

Comment:
Why do you want to move this condition above srv_read_only mode? is there any effect?

Reply:
Yes. log_enable_encryption_if_set reports an error redo log encryption is on in read only mode.

Reply:
Ah right, encryption can be enabled in read-only mode as well..